### ISSUE TITLE
Upgrade to Python 3.11

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse-catalog#855]
## Description

[Python 3.11 is finally out](https://docs.python.org/3/whatsnew/3.11.html), and with it comes a tremendous potential speedup:

> Python 3.11 is between 10-60% faster than Python 3.10. On average, we measured a 1.25x speedup on the standard benchmark suite.

There are also some other exciting changes, including [`TypedDict` optional fields improvements](https://docs.python.org/3/whatsnew/3.11.html#whatsnew311-pep655), [enriched exceptions](https://docs.python.org/3/whatsnew/3.11.html#whatsnew311-pep678), and [finer-grained exceptions](https://docs.python.org/3/whatsnew/3.11.html#whatsnew311-pep657).

We should try to get on this version as soon as possible, but unfortunately we're dependent on Airflow being Python 3.11 compatible. Work is being done upstream to accomplish this, and we should try to monitor it closely!

https://github.com/apache/airflow/issues/27292

https://github.com/apache/airflow/pull/27264

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by dhruvkb** [comment]
This issue should go in `WordPress/openverse`. We would definitely want Python 3.11 in the API and ingestion-server as well!

**Comment by AetherUnbound** [comment]
Agreed, we'll want them upgraded too! I'm assuming dependabot will actually cover those two repos next month, but I think a separate issue in this repo is reasonable since we're pretty heavily dependent on Airflow supporting 3.11 before advancing on this.
